include/chain_invoke.h

```diff
@@ -61,18 +61,38 @@ using namespace msl;
             }
         }
 
+        template<bool have_agrs, typename TRet, typename stack_args>
+        struct TypeSelector final
+        {
+            private:
+                using t1 = std::conditional_t<have_agrs, stack_args, std::tuple<TRet>>;
+                inline static constexpr bool is_ref = std::is_reference_v<TRet>;
+                using no_ref_but_pointer = std::tuple<std::add_pointer_t<std::remove_reference_t<TRet>>>;
+                using t2 = std::conditional_t<is_ref, std::tuple<no_ref_but_pointer>, std::tuple<TRet>>;
+
+            public:
+                inline static constexpr bool have_agrs_value = have_agrs;
+                inline static constexpr bool is_ref_value = is_ref;
+                using type = std::conditional_t<is_ref, t2, t1>;
+        };
+
         /**
          * @brief      The invoking step object, which also holds the tuple
          *
          * @tparam     Fx    Type of the method
          */
         template<typename Fx>
-        struct OwningInvokingStep {
-            using tuple_t    = typename function_info<Fx>::stack_args;
-            using qalified_t = typename function_info<Fx>::args;
-            using class_t    = typename function_info<Fx>::cl;
+        struct OwningInvokingStep {            
+            using TFunctionInfo = function_info<Fx>;
```

Comment:
`function_info_t`?
+ vertical alignment for "="

Reply:
done